File: backend/src/main/java/smarthost/backend/model/Review.java
package smarthost.backend.model;

import jakarta.persistence.*;
import lombok.Data;
import org.hibernate.annotations.Check;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "reviews")
@Data
public class Review {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "reservation_id", nullable = false)
    private Long reservationId;

    @Column(name = "apartment_id", nullable = false)
    private Long apartmentId;

    @Column(name = "client_id", nullable = false)
    private Long clientId;

    @Column
    @Check(constraints = "rating >= 1 AND rating <= 5")
    private Integer rating;

    @Column(columnDefinition = "TEXT")
    private String comment;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    // Relationships
    @OneToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "reservation_id", insertable = false, updatable = false)
    private Reservation reservation;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "apartment_id", insertable = false, updatable = false)
    private Apartment apartment;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "client_id", insertable = false, updatable = false)
    private User client;
}

File: frontend/src/app/models/review.ts
export interface Review {
  id: number;
  reservationId: number;
  apartmentId: number;
  clientId: number;
  rating: number; // 1-5 stars
  comment?: string;
  createdAt: string;
  
  // Related data (populated by backend)
  reservation?: {
    id: number;
    checkIn: string;
    checkOut: string;
    guestInformation: {
      firstName: string;
      lastName: string;
    };
  };
  apartment?: {
    id: number;
    name: string;
  };
  client?: {
    id: number;
    firstName: string;
    lastName: string;
  };
}

export interface CreateReviewRequest {
  reservationId: number;
  apartmentId: number;
  rating: number;
  comment?: string;
}

export interface ReviewResponse extends Review {
  // Additional response fields if needed
}

export interface PendingReviewReservation {
  id: number;
  apartmentId: number;
  apartmentName: string;
  checkIn: string;
  checkOut: string;
  reviewSubmitted: boolean;
}